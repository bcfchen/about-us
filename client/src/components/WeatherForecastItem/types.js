import PropTypes from "prop-types";
import DailyForecast from "../../models/DailyForecast";

export const propTypes = {
    forecastItem: PropTypes.instanceOf(DailyForecast).isRequired
};