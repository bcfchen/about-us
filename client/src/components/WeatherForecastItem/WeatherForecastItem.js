import React from "react";
import List from "@material-ui/core/List";
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';

const WeatherForecastItem = ({ forecastItem }) => {
    return (
        <div className="weather-forecast-item-container-even">
            <div>{forecastItem.high}</div>
            <div>{forecastItem.low}</div>
            <div>{forecastItem.day}</div>
        </div>
    );
}

export default WeatherForecastItem;