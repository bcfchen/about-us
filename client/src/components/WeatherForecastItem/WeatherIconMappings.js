import React from "react";
import Sunny from "react-icons/lib/ti/weather-sunny";
import Shower from "react-icons/lib/ti/weather-shower";
import Cloudy from "react-icons/lib/ti/weather-cloudy";
import Snow from "react-icons/lib/ti/weather-snow";

export const getWeatherIcon = weatherDesc => {
    let matchingIcon = <Sunny />;
    Object.keys(mappingRules).forEach(key => {
        let icon = mappingRules[key](weatherDesc);
        if (icon) {
            matchingIcon = icon;
            return;
        }
    });

    // return Sunny by default if no rule applies
    return matchingIcon;
};

const mappingRules = {
    sunny: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("sunny") > -1) return <Sunny />; return null; },
    downpour: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("downpour") > -1) return <Shower />; return null; },
    shower: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("downpour") > -1) return <Shower />; return null; },
    rain: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("rain") > -1) return <Shower />; return null; },
    stormy: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("stormy") > -1) return <Shower />; return null; },
    snow: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("cloud") > -1) return <Snow />; return null; },
    cloudy: weatherDesc => { if (weatherDesc.toLowerCase().indexOf("cloud") > -1) return <Cloudy />; return null; }
};
