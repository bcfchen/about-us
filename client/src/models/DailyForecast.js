import moment from "moment";

class DailyForecast {
    constructor(properties) {
        this.high = properties.Temperature.Maximum.Value;
        this.low = properties.Temperature.Minimum.Value;
        this.text = properties.Day.IconPhrase;
        this.dayOfWeek = moment(properties.Date).format("ddd");
    }
}

export default DailyForecast;