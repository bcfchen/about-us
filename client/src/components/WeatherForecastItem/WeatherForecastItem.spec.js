import React from "react";
import { shallow } from "enzyme";
import WeatherForecastItem from "./WeatherForecastItem";
import * as mappings from "./WeatherIconMappings";
import DailyForecast from "../../models/DailyForecast";
jest.mock("./WeatherIconMappings");

describe("<WeatherForecastItem/>", () => {
    let component = null,
        forecastProps = {
            Temperature: {
                Maximum: { Value: 15 },
                Minimum: { Value: 9 }
            },
            Day: { IconPhrase: "Sunny" }
        },
        defaultProps = {
            forecastItem: new DailyForecast(forecastProps)
        };

    let render = (props = defaultProps) => {
        component = shallow(<WeatherForecastItem {...props} />);
    };
    beforeEach(() => {
        mappings.getWeatherIcon.mockImplementation(() => <div className="mock-icon">mock icon</div>);
        render();
    });

    describe("on page load", () => {
        it("should render correctly", () => {
            expect(component).toMatchSnapshot();
        });

        it("should assign correct low temperature", () => {
            expect(component.find(".forecast-low").dive().dive().text()).toEqual("9");
        });

        it("should display correct icon component", () => {
            expect(component.find(".mock-icon").exists()).toEqual(true);
        });
    });
});
