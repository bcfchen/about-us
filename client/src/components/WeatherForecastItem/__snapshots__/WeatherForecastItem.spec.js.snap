// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherForecastItem/> on page load should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherForecastItem
    forecastItem={
        DailyForecast {
            "dayOfWeek": "Sat",
            "high": 15,
            "low": 9,
            "text": "Sunny",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          variant="headline"
>
          <div
                    className="mock-icon"
          >
                    mock icon
          </div>
</WithStyles(Typography)>,
        <WithStyles(Typography)
          variant="subheading"
>
          15
</WithStyles(Typography)>,
        <WithStyles(Typography)
          className="forecast-low"
          variant="subheading"
>
          9
</WithStyles(Typography)>,
        <WithStyles(Typography)
          variant="subheading"
>
          Sat
</WithStyles(Typography)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="mock-icon"
>
            mock icon
</div>,
          "variant": "headline",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock icon",
            "className": "mock-icon",
          },
          "ref": null,
          "rendered": "mock icon",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 15,
          "variant": "subheading",
        },
        "ref": null,
        "rendered": 15,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 9,
          "className": "forecast-low",
          "variant": "subheading",
        },
        "ref": null,
        "rendered": 9,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Sat",
          "variant": "subheading",
        },
        "ref": null,
        "rendered": "Sat",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            variant="headline"
>
            <div
                        className="mock-icon"
            >
                        mock icon
            </div>
</WithStyles(Typography)>,
          <WithStyles(Typography)
            variant="subheading"
>
            15
</WithStyles(Typography)>,
          <WithStyles(Typography)
            className="forecast-low"
            variant="subheading"
>
            9
</WithStyles(Typography)>,
          <WithStyles(Typography)
            variant="subheading"
>
            Sat
</WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="mock-icon"
>
              mock icon
</div>,
            "variant": "headline",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock icon",
              "className": "mock-icon",
            },
            "ref": null,
            "rendered": "mock icon",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 15,
            "variant": "subheading",
          },
          "ref": null,
          "rendered": 15,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 9,
            "className": "forecast-low",
            "variant": "subheading",
          },
          "ref": null,
          "rendered": 9,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sat",
            "variant": "subheading",
          },
          "ref": null,
          "rendered": "Sat",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
